src/Instrumentation/ExtRdKafka/src/ExtRdKafkaInstrumentation.php

```diff
@@ -0,0 +1,133 @@
+    {
+        $instrumentation = new CachedInstrumentation(
+            'io.opentelemetry.contrib.php.ext_rdkafka',
+            InstalledVersions::getVersion('open-telemetry/opentelemetry-auto-ext-rdkafka'),
+            'https://opentelemetry.io/schemas/1.25.0',
+        );
+
+        // Start root span and propagate parent if it exists in headers, for each message consumed
+        self::addConsumeHooks($instrumentation);
+        // End root span on offset commit
+        self::addCommitHooks('commit');
+        self::addCommitHooks('commitAsync');
+        // Context propagation for outbound messages
+        self::addProductionHooks();
+    }
+
+    private static function addCommitHooks($functionName)
+    {
+        hook(
+            KafkaConsumer::class,
+            $functionName,
+            null,
+            static function () {
+                $scope = Context::storage()->scope();
+                if ($scope === null) {
+                    return;
+                }
+                $scope->detach();
+                $span = Span::fromContext($scope->context());
+                $span->end();
+            }
+        );
+    }
+
+    private static function addProductionHooks()
+    {
+        hook(
+            ProducerTopic::class,
+            'producev',
+            static function (
+                ProducerTopic $exchange,
+                array $params
+            ): array {
+                // Headers are the 5th argument for the producev function
+                $carrier = [];
+                TraceContextPropagator::getInstance()->inject($carrier);
+                $propagator = Globals::propagator();
+                $propagator->inject($carrier);
+
+                if (array_key_exists('traceparent', $carrier)) {
+                    $params[4]['traceparent'] = $carrier['traceparent'];
+                }
+
+                return $params;
+            }
+        );
+    }
+
+    private static function addConsumeHooks($instrumentation)
+    {
+        hook(
+            KafkaConsumer::class,
+            'consume',
+            null,
+            static function (
+                ?KafkaConsumer $exchange,
+                array $params,
+                ?Message $message,
+                ?Throwable $exception
+            ) use ($instrumentation) : void {
+                // This is to ensure that there is data. Packages periodically poll this method in order to
+                // determine if there is a message there. If there is not, we don't want to create a span.
+                if (!$message instanceof Message || $message->err !== RD_KAFKA_RESP_ERR_NO_ERROR) {
+                    return;
+                }
+
+                $builder = $instrumentation
+                    ->tracer()
+                    // @phan-suppress-next-line PhanTypeSuspiciousStringExpression - Doesn't seem to know this has to be a string
+                    ->spanBuilder('Consumer: ' . $message->topic_name)
```

Comment:
Taking a look at the [semantic convention specification](https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/#operation-types:~:text=The%20span%20name%20SHOULD%20be%20set%20to%20the%20message%20destination%20name%20and%20the%20name%20of%20the%20operation%20being%20performed%20(as%20captured%20in%20messaging.operation.name)%20in%20the%20following%20format%3A).

> The span name SHOULD be set to the message destination name and the name of the operation being performed (as captured in messaging.operation.name) in the following format:
`<destination name> <operation name>`

> If the operation name is not specified by the messaging system, then the operation type as defined in [Operation types](https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/#operation-types) SHOULD be used:
`<destination name> <operation type>`

Reply:
Good point! I've made it be `topic_name operation_type`. This matches the semconv and to sanity check it, the Java one aswell